src/core/lib/iomgr/udp_server.c

```diff
@@ -217,14 +217,23 @@ static int prepare_socket(int fd, const struct sockaddr *addr,
     goto error;
   }
 
-  if (!grpc_set_socket_nonblocking(fd, 1) || !grpc_set_socket_cloexec(fd, 1)) {
-    gpr_log(GPR_ERROR, "Unable to configure socket %d: %s", fd,
-            strerror(errno));
+  if (grpc_set_socket_nonblocking(fd, 1) != GRPC_ERROR_NONE) {
+    gpr_log(GPR_ERROR, "Unable to set nonblocking %d: %s", fd, strerror(errno));
```

Comment:
You are discarding all the info returned by `grpc_set_socket_nonblocking`. Ideally, this server should do something like https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/tcp_server_posix.c#L347 

At the very least, print the contents of the `grpc_error` (see `grpc_error_string` at https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/error.h#L129).

Same for the rest of the changes.